specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,674 @@
+import type { HtmlGeneratorFieldData } from '../WbPlanView/LineComponents';
+import { getMappingLineProps } from '../WbPlanView/LineComponents';
+import { MappingView } from '../WbPlanView/MapperComponents';
+import type { MappingLineData } from '../WbPlanView/navigator';
+import { downloadDataSet } from '../WorkBench/helpers';
+import { PickList } from './Components';
+
+export function TableUniquenessRules({
+  container,
+  header,
+  onClose: handleClose,
+}: {
+  readonly container: SerializedResource<SpLocaleContainer>;
+  readonly header: LocalizedString;
+  readonly onClose: () => void;
+}): JSX.Element {
+  const model = React.useMemo(
+    () => strictGetModel(container.name),
+    [container]
+  );
+
+  const [tableRules = [], setTableRules, setStoredTableRules] =
+    useTableUniquenessRules(model.name);
+
+  const loading = React.useContext(LoadingContext);
+
+  const [saveBlocked, setSavedBlocked] = React.useState(false);
+
+  const [isEditing, _handleEditing, _disableEditing, toggleIsEditing] =
+    useBooleanState();
+
+  const fields = React.useMemo(
+    () => model.literalFields.filter((field) => !field.isVirtual),
+    [model]
+  );
+
+  const relationships = React.useMemo(
+    () =>
+      model.relationships.filter(
+        (relationship) =>
+          !(['one-to-many', 'many-to-many'] as RA<RelationshipType>).includes(
+            relationship.type
+          ) && !relationship.isVirtual
+      ),
+    [model]
+  );
+
+  React.useEffect(() => {
+    setSavedBlocked(
+      tableRules
+        .filter(({ duplicates }) => duplicates !== undefined)
+        .some(({ duplicates }) => duplicates!.totalDuplicates > 0)
+    );
+  }, [tableRules]);
+
+  const handleRuleValidation = React.useCallback(
+    (newRule: UniquenessRule, index: number) => {
+      const filteredRule: UniquenessRule = {
+        ...newRule,
+        fields: newRule.fields.filter(
+          (field, index) => newRule.fields.indexOf(field) === index
+        ),
+        scopes: newRule.scopes.filter(
+          (scope, index) => newRule.scopes.indexOf(scope) === index
+        ),
+      };
+      loading(
+        validateUniqueness(
+          container.name as keyof Tables,
+          filteredRule.fields as unknown as RA<never>,
+          filteredRule.scopes as unknown as RA<never>
+        ).then((duplicates) => {
+          const isNewRule = index > tableRules.length;
+          setTableRules((previous) =>
+            isNewRule
+              ? [...previous!, { rule: filteredRule, duplicates }]
+              : replaceItem(tableRules, index, {
+                  rule: filteredRule,
+                  duplicates,
+                })
+          );
+
+          return filteredRule;
+        })
+      );
+    },
+    [container.name, loading, tableRules, setTableRules]
+  );
+
+  return (
+    <Dialog
+      buttons={
+        <>
+          <Button.DialogClose>{commonText.close()}</Button.DialogClose>
+          {saveBlocked ? (
```

Comment:
this is bad UX and bad accessibility - you have that is clickable, but click won't do anything

if you look at how this works in other places in specify, when you click on the button when there are validion errors, the field with the error gets focused, and error message becomes visible

could you please refactor this a bit:
- Wrap the contents of the dialog in `<Form id={formId}>`
- Remove this Button.Danger
- Add `form={formId}` to the Submit.Save you have here - this connects the submit button to the form (does not happen automatically because the submit button is rendered outside the form)
- use `useValidation` hook inside of UniquenessRuleRow to emit a validation error for the field that is causing issues

Reply:
Fixed in https://github.com/specify/specify7/pull/4215/commits/66ad3765355550689d875e60f98a1d320d5acc52